zerver/tests/test_subs.py

```diff
@@ -565,193 +562,174 @@ def test_can_administer_channel_group_on_stream_creation(self) -> None:
-        self.assertEqual(list(stream.can_administer_channel_group.direct_members.all()), [user])
-        self.assertEqual(
-            list(stream.can_administer_channel_group.direct_subgroups.all()),
-            [moderators_system_group],
-        )
-
-        subscriptions = [{"name": "new_stream5", "description": "Fifth new stream"}]
-        internet_group = NamedUserGroup.objects.get(
-            name="role:internet", is_system_group=True, realm=realm
-        )
-        result = self.common_subscribe_to_streams(
-            user,
-            subscriptions,
-            {"can_administer_channel_group": orjson.dumps(internet_group.id).decode()},
-            allow_fail=True,
-            subdomain="zulip",
-        )
-        self.assert_json_error(
-            result,
-            "'can_administer_channel_group' setting cannot be set to 'role:internet' group.",
-        )
-
-        everyone_group = NamedUserGroup.objects.get(
-            name="role:everyone", is_system_group=True, realm=realm
-        )
-        result = self.common_subscribe_to_streams(
-            user,
-            subscriptions,
-            {"can_administer_channel_group": orjson.dumps(everyone_group.id).decode()},
-            allow_fail=True,
-            subdomain="zulip",
-        )
-        self.assert_json_error(
-            result,
-            "'can_administer_channel_group' setting cannot be set to 'role:everyone' group.",
-        )
-
-    def test_can_remove_subscribers_group_on_stream_creation(self) -> None:
+    def do_test_permission_setting_on_stream_creation(self, setting_name: str) -> None:
         user = self.example_user("hamlet")
         realm = user.realm
         self.login_user(user)
         moderators_system_group = NamedUserGroup.objects.get(
             name="role:moderators", realm=realm, is_system_group=True
         )
-        admins_system_group = NamedUserGroup.objects.get(
-            name="role:administrators", realm=realm, is_system_group=True
-        )
+
+        permission_config = Stream.stream_permission_group_settings[setting_name]
 
         subscriptions = [{"name": "new_stream1", "description": "First new stream"}]
+        extra_post_data = {}
+        extra_post_data[setting_name] = orjson.dumps(moderators_system_group.id).decode()
         result = self.common_subscribe_to_streams(
             user,
             subscriptions,
-            {"can_remove_subscribers_group": orjson.dumps(moderators_system_group.id).decode()},
+            extra_post_data,
             subdomain="zulip",
         )
         self.assert_json_success(result)
         stream = get_stream("new_stream1", realm)
-        self.assertEqual(stream.can_remove_subscribers_group.id, moderators_system_group.id)
+        self.assertEqual(getattr(stream, setting_name).id, moderators_system_group.id)
+        # Delete the created stream, so we can create a new one with
+        # same name for testing another setting.
+        stream.delete()
```

Comment:
Since we're deleting streams right after a single piece of test, should we stop using the `new_stream1`, `new_stream2` and so on in this test?